pkg/subctl/cmd/join.go

```diff
@@ -268,16 +266,30 @@ func joinSubmarinerCluster(config *rest.Config, subctlData *datafile.SubctlData)
 		err = checkOverlappingClusterCidr(globalNetworks)
 		status.End(err == nil)
 		exitOnError("Error validating overlapping ClusterCIDRs", err)
-	} else if globalNetworks[clusterID] == nil || globalNetworks[clusterID].GlobalCIDRs == nil || len(globalNetworks[clusterID].GlobalCIDRs) <= 0 {
-		// Globalnet enabled, no globalCidr configured on this cluster
-		globalCIDR, err = globalnet.AllocateGlobalCIDR(globalNetworks, subctlData)
-		status.End(err == nil)
-		status.QueueSuccessMessage(fmt.Sprintf("Allocated GlobalCIDR: %s", globalCIDR))
-		exitOnError("Globalnet failed", err)
+	} else if globalnetCIDR == "" {
+		// Globalnet enabled, GlobalCIDR not specified by the user
+		if cidrIsPreConfigured(clusterID, globalNetworks) {
+			// globalCidr already configured on this cluster
+			globalnetCIDR = globalNetworks[clusterID].GlobalCIDRs[0]
+			status.QueueSuccessMessage(fmt.Sprintf("Cluster already has GlobalCIDR allocated: %s", globalNetworks[clusterID].GlobalCIDRs[0]))
+		} else {
+			// no globalCidr configured on this cluster
+			globalnetCIDR, err = globalnet.AllocateGlobalCIDR(globalNetworks, subctlData)
+			status.End(err == nil)
+			status.QueueSuccessMessage(fmt.Sprintf("Allocated GlobalCIDR: %s", globalnetCIDR))
+			exitOnError("Globalnet failed", err)
+		}
 	} else {
-		// Globalnet enabled, globalCidr already configured on this cluster
-		globalCIDR = globalNetworks[clusterID].GlobalCIDRs[0]
-		status.QueueSuccessMessage(fmt.Sprintf("Cluster already has GlobalCIDR allocated: %s", globalNetworks[clusterID].GlobalCIDRs[0]))
+		// Globalnet enabled, globalnetCIDR specified by user
+		if cidrIsPreConfigured(clusterID, globalNetworks) {
+			// globalCidr pre-configured on this cluster
+			globalnetCIDR = globalNetworks[clusterID].GlobalCIDRs[0]
+			status.QueueFailureMessage(fmt.Sprintf("Pre-configured GlobalCIDR %s detected. Not changing it.", globalnetCIDR))
+		}
+		// globalCidr as specified by the user
+		err = checkOverlappingGlobalCidr(globalNetworks)
+		exitOnError(fmt.Sprintf("Error validating overlapping GlobalCIDRs %s", globalnetCIDR), err)
+		status.QueueSuccessMessage(fmt.Sprintf("GlobalCIDR is: %s", globalnetCIDR))
```

Comment:
I think this should be in the else part

Reply:
This should be fine too as the if block is just a warning and does not afffect the completion of command.

Reply:
Since we already found a pre-configured globalCIDR and are planning to re-use it, why do we have to check for OverlappingGlobalCIDR (for the user-supplied value)? So, this validation should be in else.

Reply:
DOne